download/oci_dowload.go

```diff
@@ -0,0 +1,369 @@
+package download
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"oras.land/oras-go/pkg/content"
+	"oras.land/oras-go/pkg/oras"
+
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/open-policy-agent/opa/bundle"
+	"github.com/open-policy-agent/opa/logging"
+	"github.com/open-policy-agent/opa/metrics"
+	"github.com/open-policy-agent/opa/plugins"
+	"github.com/open-policy-agent/opa/plugins/rest"
+	"github.com/open-policy-agent/opa/util"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/pkg/errors"
```

Comment:
I'd rather not introduce new usage of this package, we're trying to phase it out. Let's do `fmt.Errorf("yadda: %w", err)` instead.

Reply:
I used fmt.Errorf so the errors package is now removed